test/golden/artman_golden_test.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman golden tests.
+
+It runs GAPIC generation against the golden input data, can then compare the
+output layout with the expected output. This helps to capture unexpected GAPIC
+output layout change caused by either by artman, toolkit or other runtime
+dependency.
+"""
+
+import os
+import subprocess
+
+import pytest
+
+def test_library_example(clone_googleapis_for_testing=False):
+    golden_dir = os.path.dirname(os.path.realpath(__file__))
+    if clone_googleapis_for_testing:
+        subprocess.check_call([
+            'git',
+            'clone',
+            'https://github.com/googleapis/googleapis',
+            os.path.join(golden_dir, 'googleapis')
+        ])
+    else:
+        pytest.skip(
+            'Skip the golden test as the --clone_googleapis_for_testing is not '
+            'passed through py.test')
+
+    with open(os.path.join(golden_dir, 'library_example.golden')) as f:
+        expected = set(f.read().splitlines())
+
+    for lang in ['python', 'java', 'ruby', 'nodejs', 'php', 'go', 'csharp']:
+        generate_gapic_library(golden_dir, lang)
+
+    output_dir = '/output'
+    actual = []
+    for root, subdirs, files in os.walk(output_dir):
+        for f in files:
+            actual.append(os.path.join(root, f)[len(output_dir):])
+
+    print('Actual golden output')
```

Comment:
I still wish you would make this an artifact, which would allow it to be tracked over time. I will not block on it though.